server/services.py

```diff
@@ -108,6 +110,9 @@ def __init__(self, location):
     self.__pngWriter = PngWriter(self.__location)
     self.__legacyParser = Args() # TODO: inject this?
 
+    # Create the storage directory if it doesn't exist
+    Path(location).mkdir(parents=True, exist_ok=True)
+
   def __getName(self, dreamId: str, postfix: str = '') -> str:
     return f'{dreamId}{postfix}.png'
 
```

Comment:
Line 122 has swapped the metadata and legacy_metadata fields in the call to save_image_and_prompt_to_png()

Reply:
I'll update. I changed the metadata write from the prompt to the json metadata format before you introduced the `sd-meta` tag. I do successfully load both old and new format image metadata though (services.py tries to load from prompt if load from json fails).

Reply:
On (2), does the help message print after `if opt.parse_cmd(command) is None:` but before the width/height check? If so, you could move the rest of the function to a separate function, do the imports you need right after parsing the command, and then call that function. You wouldn't need all of the dependency-injected `Provide` parameters until then. I don't have the best understanding of how Python imports work yet though, or the performance implications.

You may also need to move the `generator_service.wait_for_ready()` to a later location. The model warmup should happen in the background now (in the web service, this allows it to start up immediately and queue requests and view previous results while loading the model).

Reply:
Alright @lstein I'm using the `sd-metadata` field now. I have a question though: is there any use in storing the original prompt? I don't have a prompt from the web interface, so I can't save anything for that field. We also don't appear to retrieve the result anywhere in the code base (@psychedelicious might in his web UI - is it needed?).

I'd like to stop writing the `Dream` prompt to the image in PngWriter. If we need to have a prompt for the interface, we could always generate one from the metadata, right?

Reply:
> I can confirm that the generated file has the indicated base64-encoded name, rather than the name that people expect from pngwriter.

This is intentional, though doesn't need to be final. Here's why:
1. `unique_prefix` from `PngWriter` may not work in a world where multiple results can be generated at the same time, or out of order. I had tried using a timestamp, but it also didn't provide much value.
2. Including the seed in the output was useful with iterations, but not relevant when you're not generating anything (e.g. just running an upscale on an input image).

I don't think any information is lost - seed is still in the metadata (where relevant) and files can still be sorted by creation date. Using a UUID at least ensures we shouldn't encounter conflicts, and writing it as base64 is just a bit more compact for the filename.

Reply:
> I don't have a prompt from the web interface, so I can't save anything for that field. We also don't appear to retrieve the result anywhere in the code base (@psychedelicious might in his web UI - is it needed?).

> I'd like to stop writing the Dream prompt to the image in PngWriter. If we need to have a prompt for the interface, we could always generate one from the metadata, right? 

@lstein 

I am reading the Dream prompt in my UI in an attempt to still be able to retrieve metadata from images made early on. However thinking about it now, we should just not worry about embedding the dream prompt going forward.

`Args.py` already provides a way to generate a dream prompt from metadata; I agree the client should just hav a button for this. Also allows `dream.py` to change its syntax in the future and let `Args.py` manage always making a viable dream prompt.